shell/platform/android/io/flutter/plugin/platform/SingleViewPresentation.java

```diff
@@ -236,25 +252,43 @@ private static int atMost(int measureSpec) {
     }
 
     /**
-     * Proxies a Context replacing the WindowManager with our custom instance.
+     * Proxies a Context replacing the WindowManager and InputMethodManager with our custom instance.
      */
     static class PresentationContext extends ContextWrapper {
-        private WindowManager windowManager;
-        private final WindowManagerHandler windowManagerHandler;
-
-        PresentationContext(Context base, WindowManagerHandler windowManagerHandler) {
+        private @Nullable
+        final WindowManagerHandler windowManagerHandler;
+        private @NonNull
+        final InputMethodManager inputMethodManager;
```

Comment:
Is it safe to cache this here? is it possible that the IMM for FlutterView is changed after we cached it? e.g FlutterView was moved to a different window (I actually won't be surprised if it's common for code that was written prior to Q to assume the IMM doesn't change, so not sure if there's any fail safe that mitigates it already).

Reply:
I'm not sure we could extend presentation without calling it's constructor (which uses `createPresentationContext`).

Reply:
For the new embedding the plan is to have the PlatformViewsController aware of the current FlutterView (@matthew-carroll is doing this I believe). When that is done it should be easy to use the current FlutterView's IMM. As this thing should land soon let's leave a TODO here for now.

Reply:
Okay, added a TODO for now.